plugins/nodes/src/js/components/NodeBreadcrumbs.js

```diff
@@ -19,22 +19,16 @@ const NodeBreadcrumbs = ({ nodeID, taskID, taskName, unitID }) => {
       </BreadcrumbTextContent>
     </Breadcrumb>
   ];
-  let node;
+  const node = useNode(nodeID);
```

Comment:
Would it make sense passing the node as a prop to the component instead?

Reply:
I agree with @nLight. I think ideally you would limit the components who can get data from other places, like stores.  Even down below we do the same for Units.  If we could say something like, `only top level components get data from stores, everything below it get its data from props` we could be much smarter about renders and generally have a better sense of how the data flows since something as nested as Breadcrumb isn't going out to a store to get some random data.

This is more a general comment so not something we can necessarily solve in this PR, but I think it would be worth it to sit down and think about our general arch rules like this so that when writing new code, such as this PR, we can start putting in place better fundamentals.

Reply:
Yes I definitely agree that the ideal case would be top level components retrieving any data necessary and passing down to other components. What we could do in this case, is use the `useNode` hook in components that need to retrieve nodes by ID. In some cases we can already just pass the entire node to `NodeBreadcrumbs` (for example, if we moved `useNode` to `NodeDetailPage`, we can pass the entire node to `NodeBreadcrumbs`). In other cases though, we will need to implement this data fetching at a higher level (for example, `NodesTaskDetailPage` only has the nodeId from the route params, so we would need to fetch the node in this component using `useNode` and pass it to `NodeBreadcrumbs`)